PMMain/Midas/MidasGUI/SCDDigester.cpp

```diff
@@ -115,10 +117,15 @@ void SCDDigester::digestKeyboardData(commandData nextCommand)
             //threadHandle->UpdateGUIToChangeWheels TODO
             scdHandle->setKybdGuiSel(kybdGUISel);
             break;
+
         case kybdGUICmds::SELECT:
+            if (kybdGUISel % 4 == 0)
+                kybrdCtrl->setKeyChar(kybrdRingData[kybdGUISel].getRingInVector()[scdHandle->getKeySelectAngle() / (360 / kybrdRingData[kybdGUISel].getRingInVector().size())].main);
```

Comment:
You should probably split this up into more than one line. If anything goes wrong, it will be hard to tell what calculation is incorrect. Also, it may be worth checking that the indices are valid.